api/main.py

```diff
@@ -147,10 +147,22 @@ def get_password_hash(password: Password):
 # -----------------------------------------------------------------------------
 # Nodes
 
-@app.get('/node/{node_id}', response_model=Node)
-async def get_node(node_id: str):
+@app.get('/node/{node_id}')
+async def get_node(node_id: str, kind: str = "node"):
     """Get node information from the provided node id"""
-    return await db.find_by_id(Node, node_id)
```

Comment:
It is a bit of a shame to be dropping `response_model=None` but I'm not sure what's best here.

One option would be to have some generic `data` field in the Node model so that other kinds of nodes could add things while still being a Node. That means the extra things would not be going through Pydantic validation, but it's more flexible in some ways.

Another option is to return a base `Node` object, but then we would be missing the extra fields from sub-class such as `Regression`.

I guess this is also something to take into account while reviewing how models / collections / kinds are supposed to be implemented and use by the API.

Reply:
The better approach is to use `Union`.

Reply:
Erm, Union should be about data structures with variants. Here we have a plain object inheritance with just some added fields to the child class.